packages/map-toolkit/src/hover-coordinate/use-hover-coordinate.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { useOn } from '@accelint/bus/react';
+import { coordinateSystems, createCoordinate } from '@accelint/geo';
+import { useContext, useEffect, useState } from 'react';
+import { MapEvents } from '../deckgl/base-map/events';
+import { MapContext } from '../deckgl/base-map/provider';
+import type { UniqueId } from '@accelint/core';
+import type { MapHoverEvent } from '../deckgl/base-map/types';
+
+/**
+ * Supported coordinate format types for displaying map coordinates.
+ *
+ * @typedef {'dd' | 'ddm' | 'dms' | 'mgrs' | 'utm'} FormatTypes
+ * @property dd - Decimal Degrees (e.g., "45.50000000 E / 30.25000000 N")
+ * @property ddm - Degrees Decimal Minutes (e.g., "45 30 E / 30 15 N")
+ * @property dms - Degrees Minutes Seconds (e.g., "45 30 0 E / 30 15 0 N")
+ * @property mgrs - Military Grid Reference System
+ * @property utm - Universal Transverse Mercator
+ */
+export type FormatTypes = 'dd' | 'ddm' | 'dms' | 'mgrs' | 'utm';
+
+const create = createCoordinate(coordinateSystems.dd, 'LONLAT');
+
+const MAX_LONGITUDE = 180;
+const LONGITUDE_RANGE = 360;
+const COORDINATE_PRECISION = 8;
+const prepareCoord = (coord: [number, number]) => {
+  // Normalize longitude to -180 to 180 range (handles any wraparound including multi-revolution values)
+  let lon = coord[0];
+  lon =
+    ((((lon + MAX_LONGITUDE) % LONGITUDE_RANGE) + LONGITUDE_RANGE) %
+      LONGITUDE_RANGE) -
+    MAX_LONGITUDE;
+
+  const lat = coord[1];
+  const lonStr = `${Math.abs(lon).toFixed(COORDINATE_PRECISION)} ${lon < 0 ? 'W' : 'E'}`;
+  const latStr = `${Math.abs(lat).toFixed(COORDINATE_PRECISION)} ${lat < 0 ? 'S' : 'N'}`;
+
+  return `${lonStr} / ${latStr}`;
+};
+
```

Comment:
pull the coordinate check into a function

```suggestion
function isValidCoordinate(value: number[]): value is [number, number] {
  return (
    Array.isArray(value) &&
    value.length === 2 &&
    value.every((Number.isFinite))
  );
}
```